src/azure-cli-core/azure/cli/core/commands/command_operation.py

```diff
@@ -0,0 +1,509 @@
+        cmd_args['updated'] = CLICommandArgument(
+            'updated', options_list=['--updated'], action='store_true', arg_group=group_name,
+            help="wait until updated with provisioningState at 'Succeeded'"
+        )
+        cmd_args['exists'] = CLICommandArgument(
+            'exists', options_list=['--exists'], action='store_true', arg_group=group_name,
+            help="wait until the resource exists"
+        )
+        cmd_args['custom'] = CLICommandArgument(
+            'custom', options_list=['--custom'], arg_group=group_name,
+            help="Wait until the condition satisfies a custom JMESPath query. E.g. "
+                 "provisioningState!='InProgress', "
+                 "instanceView.statuses[?code=='PowerState/running']"
+        )
+        return [(k, v) for k, v in cmd_args.items()]
+
+    def description_loader(self):
+        """ Callback function of CLICommand description_loader """
+        return self.load_op_handler_description()
+
+
+class ShowCommandOperation(BaseCommandOperation):
+
+    def __init__(self, ctx, op_path, **kwargs):
+        if not isinstance(op_path, string_types):
+            raise TypeError("operation must be a string. Got '{}'".format(op_path))
+        super(ShowCommandOperation, self).__init__(ctx, **kwargs)
+        self.op_path = op_path
+
+    def handler(self, command_args):
+        """ Callback function of CLICommand handler """
+        from azure.cli.core.commands.arm import show_exception_handler, EXCLUDED_NON_CLIENT_PARAMS
+
+        op = self.get_op_handler(self.op_path)
+        getter_args = dict(extract_args_from_signature(op, excluded_params=EXCLUDED_NON_CLIENT_PARAMS))
+
+        self.cmd = command_args.get('cmd') if 'cmd' in getter_args else command_args.pop('cmd')
+
+        client_arg_name = self.resolve_client_arg_name(self.op_path)
+        try:
+            client = self.client_factory(self.cli_ctx) if self.client_factory else None
+        except TypeError:
+            client = self.client_factory(self.cli_ctx, command_args) if self.client_factory else None
+
+        if client and (client_arg_name in getter_args):
+            command_args[client_arg_name] = client
+
+        op = self.get_op_handler(self.op_path)  # Fetch op handler again after cmd property is set
+        try:
+            return op(**command_args)
+        except Exception as ex:  # pylint: disable=broad-except
+            show_exception_handler(ex)
+
+    def arguments_loader(self):
+        """ Callback function of CLICommand arguments_loader """
+        cmd_args = self.load_getter_op_arguments(self.op_path)
+        return [(k, v) for k, v in cmd_args.items()]
+
+    def description_loader(self):
+        """ Callback function of CLICommand description_loader """
+        op = self.get_op_handler(self.op_path)
+        return self.load_op_handler_description(op)
+
+
+class GenericUpdateCommandOperation(BaseCommandOperation):     # pylint: disable=too-many-instance-attributes
```

Comment:
could you make all operations in Alphabetical order?

Reply:
Good advice